### PR TITLE
Handle array attributes in hash/equality methods.

### PR DESCRIPTION
**Pull request by chrisgibson**
# Problem

Array attributes are not handled in `#==`, `#eql?`, and `#hash`. A `TypeError` is raised when calling `send` with the attribute within `#==` and `hash`.

# Solution

Unwrap array attribute names (similar to `.initialize`) in the `#==` and `#hash` methods.

### CODE REVIEW

File: lib/immutable-struct.rb [hunk 1]
```diff
@@ -88,7 +88,12 @@ def self.new(*attributes,&block)
 
       define_method(:==) do |other|
         return false unless other.is_a?(klass)
-        attributes.all? { |attribute| self.send(attribute) == other.send(attribute) }
+        attributes.all? do |attribute|
+          if attribute.kind_of?(Array) and attribute.size == 1
```
Comment (davetron5000) [inline lib/immutable-struct.rb]: Is this because `attribute` is using that syntax `[:widgets]`?

**Comment by sipple** [comment]
Curious about the failure in travis for `rbx-2`, as it's passing on everything else. I'm not actually sure what `rbx-2` is, but it seems to have been failing for a while. Is this a long term issue that we're not dealing with?

File: lib/immutable-struct.rb (hunk 1 above)
Comment (chrisgibson) [inline lib/immutable-struct.rb]: That's right... for array attributes, we're trying to `send` that to the instance `send([:widgets])`...

**Comment by davetron5000** [comment]
rbx is rubinius.  It says there are no binaries to download, so maybe that's not the right name for Rubinius?  Latest is version 3, so let's try `rbx-3` and if that doesn't work, let's just remove it.

**Comment by sipple** [comment]
@davetron5000 👍

**Pull request merged**